RecoLocalTracker/SiPixelClusterizer/plugins/SiPixelRawToClusterGPUKernel.cu

```diff
@@ -397,14 +350,15 @@ namespace pixelgpudetails {
   {
     //if (threadIdx.x==0) printf("Event: %u blockIdx.x: %u start: %u end: %u\n", eventno, blockIdx.x, begin, end);
 
-    auto gIndex  = threadIdx.x + blockIdx.x * blockDim.x;
-    xx[gIndex]   = 0;
-    yy[gIndex]   = 0;
-    adc[gIndex]  = 0;
-    bool skipROC = false;
+  int32_t first = threadIdx.x + blockIdx.x*blockDim.x;
+  for (int32_t iloop=first, nend=wordCounter; iloop<nend; iloop+=blockDim.x*gridDim.x) { 
```

Comment:
Is the `iloop+=blockDim.x*gridDim.x` more about future-proofing than a necessity? (since the kernel call side, i..e number of threads and blocks are not changed)

Reply:
is equivalent to the previous ```if (gIndex < wordCounter) ``` and more robust as it supports any number of blocks
Alll kernels will be changed to this style

Reply:
So future-proofing, good.